lib/src/neu/verifications/contract-mismatcher.ts

```diff
@@ -0,0 +1,223 @@
+      }, {})
+    };
+
+    const validateFn = jsv.compile(schema);
+    const valid = validateFn(body);
+    if (valid) {
+      return ok([]);
+    } else {
+      if (!validateFn.errors) {
+        return err(
+          new Error(
+            `Body Validation reaches unexpected error for ${body} with contract body ${contractBodyTypeToVerifyWith}`
+          )
+        );
+      }
+      return ok(this.errorObjectMapper(validateFn.errors));
+    }
+  }
+
+  private errorObjectMapper(array: ErrorObject[]): Mismatch[] {
+    return array.map(e => {
+      return new Mismatch(
+        e.message ||
+          `JsonSchemaValidator encountered an unexpected error for ${e.data}.`
+      );
+    });
+  }
+
+  private getRequestBodyTypeOnContractEndpoint(
+    endpoint: Endpoint
+  ): Result<Type, Error> {
+    if (!endpoint.request || !endpoint.request.body) {
+      return err(
+        new Error(
+          `Request body on endpoint path ${endpoint.path}:${endpoint.method} does not exist.`
+        )
+      );
+    } else {
+      return ok(endpoint.request.body.type);
+    }
+  }
+
+  private getResponseBodyTypeOnContractEndpoint(
+    endpoint: Endpoint,
+    userInputStatusCode: number
+  ): Result<Type | undefined, Error> {
+    if (endpoint.responses.length === 0) {
+      return err(
+        new Error(
+          `There is no response defined on endpoint path ${endpoint.path}:${endpoint.method}.`
+        )
+      );
+    }
+    for (const response of endpoint.responses) {
+      if (response.status === userInputStatusCode) {
+        if (!response.body) {
+          // If contract response body does not have a type, then there is no mismatch.
+          return ok(undefined);
+        }
+        return ok(response.body.type);
+      }
+    }
+    return this.getDefaultResponseBodyTypeOnContractEndpoint(endpoint);
+  }
+
+  private getDefaultResponseBodyTypeOnContractEndpoint(
+    endpoint: Endpoint
+  ): Result<Type, Error> {
+    if (!endpoint.defaultResponse || !endpoint.defaultResponse.body) {
+      return err(
+        new Error(
+          `Default response does not exist on ${endpoint.path}:${endpoint.method}.`
+        )
+      );
+    }
+    return ok(endpoint.defaultResponse.body.type);
+  }
+
+  private getEndpointByRequest(
+    userInputRequest: UserInputRequest
+  ): Result<Endpoint, Error> {
+    for (const endpoint of this.contract.endpoints) {
+      if (
+        this.isMatchedToContractPath(userInputRequest.path, endpoint.path) &&
+        endpoint.method === userInputRequest.method
+      ) {
+        return ok(endpoint);
+      }
+    }
+    return err(
+      new Error(
+        `Endpoint ${userInputRequest.path} with Http Method of ${userInputRequest.method} does not exist under the specified contract.`
+      )
+    );
+  }
+
+  private isMatchedToContractPath(
+    userInputPath: string,
+    contractPath: string
+  ): boolean {
+    const replacedContractPathPattern = contractPath.replace(/:.*\//g, ".*/");
```

Comment:
```suggestion
    const replacedContractPathPattern = contractPath.replace(/:[^\/]*/g, ".+");
```
`"/users/:userId".replace(/:.*\//g, ".*/") === "/users/:userId"`